### PR TITLE
Add book walking section to `DetailPetSitterView`

### LINKED ISSUE #83: Add `ListOfAvailableWalks` for owner flow
## This task is a part of the issue:
 - #86

## Depends on tasks:
- #72 

## Description
Add the new section to the `DetailedSitterView` with an available walking list for selected sitter.
When the user taps on one of the available slots a new screen (BookingWalkingView) will be presented where the user can enter info about himself to book the selected slot.

## UI Mocks
Screenshot 2024-01-23 at 11 36 46

You are more than welcome to choose your design, but please follow App Style guidelines.

### PR DESCRIPTION
**Pull request by NadzeyaShpakouskaya**
## Summary
This PR is a part of the work to resolve #83 
Add the temporary method in `DetailetPetSitterViewModel` to simulate the fetching process where dummy data was returned.

## UI Preview (Screenshots)
<details>
  <summary>Spoiler</summary>
Screenshot 2024-03-05 at 12 10 42
</details>

## Changes
- Add a section with the available slots for the selected date
- Add the `BookingSlotView` stub for navigation

## Checks to complete
- [x] The PR title has a conventional commit style (short and descriptive).
- [x] The PR description includes essential information that helps understand the purpose of the changes.
- [x] I've built and run my changes, and no warnings or errors occurred.
- [x] All existing tests passed with my changes.
- [x] My code follows our style guide.
- [x] I've performed a self-review of my code.
- [x] Every non-private interface is documented properly.
- [x] I've added tests for my code (if it makes sense).
- [x] I've updated `CHANGELOG.md` file.
- [x] PR has assignee, reviewers, milestone, it is properly linked to the project and to the issue.

### CODE REVIEW

File: Woof/Woof/View/OwnerFlow/DetailedPetSitterView/DetailedPetSitterView.swift [hunk 1]
```diff
@@ -20,15 +20,44 @@ struct DetailPetSitterView: View {
...
+                .padding(.bottom)
+
+            HStack {
+                Text(slotHeaderLabel)
+                Spacer()
+                DatePicker("",
+                           selection: $viewModel.selectedDate,
+                           in: Date.now...,
+                           displayedComponents: .date).labelsHidden()
+                    .font(.system(size: AppStyle.FontStyle.heading.size))
```
Comment (kirpichonok) [inline Woof/Woof/View/OwnerFlow/DetailedPetSitterView/DetailedPetSitterView.swift:37]: nit: This looks more readable:
```suggestion
                           displayedComponents: .date)
                           .labelsHidden()
                           .font(.system(size: AppStyle.FontStyle.heading.size))
```

File: Woof/Woof/View/OwnerFlow/DetailedPetSitterView/DetailedPetSitterView.swift [hunk 2]
```diff
@@ -20,15 +20,44 @@ struct DetailPetSitterView: View {
...
+                        await viewModel.fetchWalkings()
+                    }
+                }
+                .buttonStyle(CapsuleWithWhiteText())
+            }.padding()
+            if viewModel.isLoading {
+                ProgressView()
+            } else {
+                WalkingSlotsView(walkings: $viewModel.walkings)
+            }
```
Comment (kirpichonok) [inline Woof/Woof/View/OwnerFlow/DetailedPetSitterView/DetailedPetSitterView.swift:51]: nit: This seems to look more readable:
```suggestion
            }
            .padding()

            if viewModel.isLoading {
                ProgressView()
            } else {
                WalkingSlotsView(walkings: $viewModel.walkings)
            }
```

File: Woof/Woof/View/OwnerFlow/DetailedPetSitterView/WalkingSectionView/WalkingSlotCardView.swift [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        )
+    }
+}
+
+#Preview {
+    WalkingSlotCardView(
+        price: 12.25,
+        startDate: Date.now,
+        endDate: Date.now + 1200
+    )
```
Comment (kirpichonok) [inline Woof/Woof/View/OwnerFlow/DetailedPetSitterView/WalkingSectionView/WalkingSlotCardView.swift:50]: nit: I think in this way, it's easier to see the form of the view.
```suggestion
     ZStack {
        Color.gray
        WalkingSlotCardView(
            price: 12.25,
            startDate: Date.now,
            endDate: Date.now + 1200
        )
    }
```

File: Woof/Woof/View/OwnerFlow/DetailedPetSitterView/WalkingSectionView/WalkingSlotCardView.swift [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        .background(Color.App.white)
+        .clipShape(
+            RoundedRectangle(
+                cornerRadius: AppStyle.UIElementConstant.cornerRadius
+            )
+        ).font(
+            Font.system(
+                size: AppStyle.FontStyle.footnote.size
+            )
+        )
```
Comment (kirpichonok) [inline Woof/Woof/View/OwnerFlow/DetailedPetSitterView/WalkingSectionView/WalkingSlotCardView.swift:41]: nit: we can use here the language ability to infer
```suggestion
        ).font(
            .system(
                size: AppStyle.FontStyle.footnote.size
            )
        )
```

Comment (kirpichonok) [inline Woof/Woof/View/OwnerFlow/DetailedPetSitterView/WalkingSectionView/WalkingSlotCardView.swift:41]: We can set the font size in the place of the view usage and be not linked to the font size here. WDYT?
```suggestion
        )
```

File: Woof/Woof/ViewModel/Owner/DetailPetSitterViewModel.swift [hunk 5]
```diff
@@ -46,5 +55,16 @@ final class DetailSitterViewModel: ObservableObject {
...
+    /// Fetches data about available walking slots from the remote server.
+    func fetchWalkings() async {
+        // Temporary simulation of loading process: load dummy data after 1 sec delay.
+        Task { @MainActor in
+            isLoading = true
+            try await Task.sleep(nanoseconds: 1_000_000_000)
+            walkings = Walking.Dummy.bulkDummyWalkings
+            isLoading = false
+        }
+    }
```
Comment (kirpichonok) [inline Woof/Woof/ViewModel/Owner/DetailPetSitterViewModel.swift:67]: It seems the same
```suggestion
   @MainActor func fetchWalkings() async {
            isLoading = true
            // Temporary simulation of loading process: load dummy data after 1 sec delay.
            try? await Task.sleep(nanoseconds: 1_000_000_000)
            walkings = Walking.Dummy.bulkDummyWalkings
            isLoading = false
    }
```

**Review by kirpichonok — CHANGES_REQUESTED** [review]
Great job! I have some little suggestions 🙂

File: Woof/Woof/View/OwnerFlow/DetailedPetSitterView/DetailedPetSitterView.swift (hunk 1 above)
Comment (NadzeyaShpakouskaya) [reply Woof/Woof/View/OwnerFlow/DetailedPetSitterView/DetailedPetSitterView.swift]: Fixed.

File: Woof/Woof/View/OwnerFlow/DetailedPetSitterView/DetailedPetSitterView.swift (hunk 2 above)
Comment (NadzeyaShpakouskaya) [reply Woof/Woof/View/OwnerFlow/DetailedPetSitterView/DetailedPetSitterView.swift]: Yes, fixed

File: Woof/Woof/View/OwnerFlow/DetailedPetSitterView/WalkingSectionView/WalkingSlotCardView.swift (hunk 4 above)
Comment (NadzeyaShpakouskaya) [reply Woof/Woof/View/OwnerFlow/DetailedPetSitterView/WalkingSectionView/WalkingSlotCardView.swift]: Fixed.

File: Woof/Woof/ViewModel/Owner/DetailPetSitterViewModel.swift (hunk 5 above)
Comment (NadzeyaShpakouskaya) [reply Woof/Woof/ViewModel/Owner/DetailPetSitterViewModel.swift]: Your version is more readable, fixed.

File: Woof/Woof/View/OwnerFlow/DetailedPetSitterView/WalkingSectionView/WalkingSlotCardView.swift (hunk 3 above)
Comment (NadzeyaShpakouskaya) [reply Woof/Woof/View/OwnerFlow/DetailedPetSitterView/WalkingSectionView/WalkingSlotCardView.swift]: Added a background to preview.

File: Woof/Woof/View/OwnerFlow/DetailedPetSitterView/WalkingSectionView/WalkingSlotCardView.swift (hunk 4 above)
Comment (NadzeyaShpakouskaya) [reply Woof/Woof/View/OwnerFlow/DetailedPetSitterView/WalkingSectionView/WalkingSlotCardView.swift]: Moved it to `WalkingSlotsView`.

**Review by kirpichonok — APPROVED** [review]
LGTM

**Pull request merged**